### DISCUSSION TITLE
daisyui in shadow dom

### DISCUSSION THREAD

**Discussion by edman** [saadeghi/daisyui#1659] [Q&A]
As in https://github.com/saadeghi/daisyui/issues/687 I'm trying to use daisyui in a browser extension.

The advice on that issue is to use `[data-theme]` in a root div, but it did not work for me out of the box because CSS variables are set in the `:root` selector:

```
$ cat dist/app.css | grep -A4 ":root"
:root,
[data-theme] {
  background-color: hsla(var(--b1) / var(--tw-bg-opacity, 1));
  color: hsla(var(--bc) / var(--tw-text-opacity, 1));
}
--
:root {
  color-scheme: dark;
  --pf: 198.44 93.204% 47.686%;
  --sf: 234.45 89.474% 59.137%;
  --af: 328.85 85.621% 56%;
```

My solution was to add a build step to modify the selector to `:root, [data-theme] { ... }` in the second rule.

This works but feels like it can break at any update. Is there a better way to make daisyui work in shadow dom?

**Comment by mubaidr** [comment]
Any suggestions on this topic?

**Reply by saadeghi** [reply]
Using `root` config you can set the CSS variables to a specific selector.  
https://play.tailwindcss.com/5M2AcdpoP6?file=css  
https://daisyui.com/docs/config/#root
